web/src/components/DatePicker.tsx

```diff
@@ -132,34 +89,35 @@ const CustomDatePicker = ({
 
   return (
     <DateContainer>
-      <StyledInputWrapper>
-        <StyledInput
-          open={isCalendarOpen}
-          value={value}
-          popperPlacement='bottom'
-          selected={isValidJsDate(date?.toJSDate() ?? null) ? date?.toJSDate() : null}
-          onSelect={e => {
-            setValue(DateTime.fromJSDate(e ?? new Date()).toFormat('dd.MM.yyyy'))
-            setDatePickerError('')
+      <LocalizationProvider dateAdapter={CustomAdapterLuxon} adapterLocale={languageCode}>
+        <DatePicker
+          value={date}
+          onChange={newValue => {
+            if (newValue) {
+              handleDateChange(newValue.toJSDate())
+              setValue(newValue.toFormat('dd.MM.yyyy'))
+              setDatePickerError('')
+            }
+          }}
```

Comment:
🔧 Is this custom logic actually necessary? It seems to me like this works just fine as well:
```suggestion
          onChange={setDate}
```